### PR TITLE
Issue 3296

### LINKED ISSUE #3296: LexicalPreservation bug for array brackets
Hello all, thank you for developing such a great tool. However, I found a bug when using `LexicalPreservation` tool when I try to add an annotation to a class field which has the type `array` and the brackets `[]` are behind the parameter's name as shown in the example below.

```java
class Test{
   String[] allTest;
}
```
Please note that the example below works fine and the exception happens only when the brackets are behind field name. 

```java
class Test{
   String allTest[];
}
```

I also provided the code below that will generate the same exception mentioned above when executed:
```java
String code = "public class Test { String[] allTest; }";
CompilationUnit cu = LexicalPreservingPrinter.setup(StaticJavaParser.parse(code));
Optional<ClassOrInterfaceDeclaration> clazzOptional = cu.getClassByName("Test");
assertTrue(clazzOptional.isPresent());
ClassOrInterfaceDeclaration clazz = clazzOptional.get();
clazz.getMembers().forEach(
        bodyDeclaration ->
                bodyDeclaration.ifFieldDeclaration(
                        fieldDeclaration -> {
                            NodeList<VariableDeclarator> vars =
                                    fieldDeclaration.asFieldDeclaration().getVariables();
                            for (VariableDeclarator v : vars) {
                                if (v.getName().toString().equals("allTest")) {
                                    fieldDeclaration.addMarkerAnnotation("Nullable");
                                    break;
                                }
                            }
                        }));
String changed = LexicalPreservingPrinter.print(cu);
```

Here is the error I get when executing the code above:

```java
java.lang.UnsupportedOperationException: kept child(VariableDeclarator) vs "["   <101>   (line 1,col 27)-(line 1,col 27)

	at com.github.javaparser.printer.lexicalpreservation.Difference.applyKeptDiffElement(Difference.java:513)
```

With best regards, 

Nima Karimipour

### PR DESCRIPTION
**Pull request by nimakarimipour**
Fixes #3296

### CODE REVIEW

**Comment by nimakarimipour** [comment]
@jlerbsc I believe this is ready for a review !

File: javaparser-core-generators/pom.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <parent>
         <artifactId>javaparser-parent</artifactId>
         <groupId>com.github.javaparser</groupId>
-        <version>3.24.2-SNAPSHOT</version>
+        <version>3.22.2-SNAPSHOT</version>
```
Comment (jlerbsc) [inline javaparser-core-generators/pom.xml]: Why did you modify the java parser version?

**Comment by jlerbsc** [comment]
Can you amend your commits to have a clean history?

File: javaparser-core-generators/pom.xml (hunk 1 above)
Comment (nimakarimipour) [reply javaparser-core-generators/pom.xml]: This was for having two versions on my local maven, in the final version of this PR every version is reverted to its original one.

**Comment by nimakarimipour** [comment]
> Can you amend your commits to have a clean history?

Sure, I will.

**Pull request closed without merging**